src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -185,6 +230,7 @@ private static void loadNativeLibrary() {
         }
     }
 
+    private static native boolean createReferenceIndex(String referenceName, String indexPrefix, String algorithmName);
```

Comment:
Perhaps  someone sometime would be only interested in indexing a reference? We can use Java code to unite both operations (indexing and image-loading), for example to create the index on the fly when the image is requested but there is no index for that reference. 

In any case I rather keep the C code to a bare minimum specially since we currently have no way to recover or properly report errors in libbwa that end up in an exit/abort or seg-fault.